EOS-16559: Re-align existing pillar data for R2

ypise opened the pull request:
Signed-off-by: Yashodhan Pise <yashodhan.pise@seagate.com>

sradhanand-pati commented on Cortx-ProvisionerQuickstartGuide.md:
```diff
@@ -641,23 +641,23 @@ In such scenarios the destroy may get stuck somewhere due to some unknown reason
             		id: storage_node_1            # equivalent to fqdn for server node
             		type: RBOD                    # Type of enclosure. E.g. RBOD
-              		primary_mc:
+              		primary
```
i guess you missed a ':' here

ypise replied:
Yes

nitin-seagate commented on srv/components/system/network/mgmt/bonded/init.sls:
```diff
@@ -15,7 +15,7 @@
 #
 
-{%- if (pillar['cluster'][node]['network']['mgmt_nw']['ipaddr']) and (not pillar['cluster'][grains['id']]['network']['mgmt_nw']['gateway']) %}
+{%- if (pillar['cluster'][node]['network']['mgmt'ipaddr']) and (not pillar['cluster'][grains['id']]['network']['mgmmgmtteway']) %}
```
This is incorrect, can you check if we need to update:
['mgmt]['public_ip']
['mgmt']['gateway']

nitin-seagate commented on srv/components/system/network/mgmt/bonded/init.sls:
```diff
@@ -29,18 +29,18 @@ Ensure bonding config for management bond:
 # Setup network for management interfaces
-{% for iface in pillar['cluster'][node]['network']['mgmt_nw']['iface'] %}
-{{ iface }}:
+{% for interface in pillar['cluster'][node]['network']['mgmt'interface'] %}
```
Guess this should be ['mgmt']['interfaces']

nitin-seagate commented on srv/components/system/network/mgmt/bonded/init.sls:
```diff
@@ -55,14 +55,14 @@ Setup mgmt0 bonding:
     - mode: active-backup
     - miimon: 100
-{% if pillar['cluster'][node]['network']['mgmt_nw']['ipaddr'] %}
+{% if pillar['cluster'][node]['network']['mgmt'ipaddr'] %}
```
['mgmt]['public_ip']

nitin-seagate commented on srv/components/system/network/mgmt/bonded/init.sls:
```diff
@@ -55,14 +55,14 @@ Setup mgmt0 bonding:
-    - ipaddr: {{ pillar['cluster'][node]['network']['mgmt_nw']['ipaddr'] }}
-{% if pillar['cluster'][node]['network']['data_nw']['netmask'] %}
-    - netmask: {{ pillar['cluster'][node]['network']['data_nw']['netmask'] }}
+    - ipaddr: {{ pillar['cluster'][node]['network']['mgmt'ipaddr'] }}
```
['mgmt]['public_ip']

nitin-seagate commented on srv/components/system/network/mgmt/bonded/init.sls:
```diff
@@ -55,14 +55,14 @@ Setup mgmt0 bonding:
-{% if pillar['cluster'][node]['network']['data_nw']['gateway'] %}
-    - gateway: {{ pillar['cluster'][grains['id']]['network']['mgmt_nw']['gateway'] }}
+{% if pillar['cluster'][node]['network']['data']['gateway'] %}
+    - gateway: {{ pillar['cluster'][grains['id']]['network']['mgmt'gateway'] }}
```
['mgmt]['gateway']

nitin-seagate commented on srv/components/system/network/mgmt/public/init.sls:
```diff
@@ -16,16 +16,16 @@
 {% else %}
 
-{% set mgmt_if = pillar['cluster'][node]['network']['mgmt_nw']['iface'][0] %}
+{% set mgmt_if = pillar['cluster'][node]['network']['mgmt']['interface'][0] %}
```
Guess this should be ['interfaces']

nitin-seagate commented on srv/components/system/network/mgmt/service_port/init.sls:
```diff
@@ -16,16 +16,16 @@
 
 {% set node = grains['id'] %}
-{% if pillar['cluster'][node]['network']['mgmt_nw']['iface'][1] is defined %}
+{% if pillar['cluster'][node]['network']['mgmt']['interface'][1] is defined %}
```
Guess this should be ['interfaces']

nitin-seagate commented on srv/components/system/network/mgmt/service_port/init.sls:
```diff
@@ -16,16 +16,16 @@
   network.managed:
-    - name: {{ pillar['cluster'][node]['network']['mgmt_nw']['iface'][1] }}
-    - device: {{ pillar['cluster'][node]['network']['mgmt_nw']['iface'][1] }}
+    - name: {{ pillar['cluster'][node]['network']['mgmt']['interface'][1] }}
```
Guess this should be ['interfaces']

nitin-seagate commented on test/api/python/integration/test_api_inner.py:
```diff
@@ -310,10 +310,10 @@ def test_set_network():
     pillar_nw_primary_mgmt_ip = pillar[
         'srvnode-1'
-    ]['cluster']['srvnode-1']['network']['mgmt_nw']['ipaddr']
+    ]['cluster']['srvnode-1']['network']['mgmt']['ipaddr']
```
Guess this should be ['mgmt_ip']['public_ip']

nitin-seagate commented on test/api/python/provisioner/test_commands.py:
```diff
@@ -67,7 +67,7 @@ def storage_enclosure_data(mocker):
         "storage_enclosure": {
             "controller": {
-                "primary_mc": {
+                "primary": {
```
Please handle secondary as well?

ypise replied:
Done

nitin-seagate reviewed (changes requested):
Summary of requested changes:
1. Change keyword **ipaddr** to either **public_ip** or **private_ip** (based on mgmt/data network)
2.  For data network ensure **public_ip** and **private_ip** info is added
3.  For storage enclosure ensure **primary** and **secondary** controller info is added
4. Change reference of list of interface with keyword **interfaces**
5. Ensure **machine_id** is added for server node info
6. Ensure **server_nodes** and **storage_sets** mappings are correct

ypise commented on api/python/provisioner/commands/check.py:
```diff
@@ -1053,10 +1053,10 @@ def _public_data_ip(*, args: str) -> Union[CheckEntry, List[CheckEntry]]:
+                    f"cluster/{node}/network/data/interface")
 
-                ifc = [ifc for ifc in iface if 's0f0' in ifc]
+                ifc = [ifc for ifc in interface if 's0f0' in ifc]
```
Yes

ypise commented on api/python/provisioner/commands/configure_setup.py:
```diff
@@ -222,7 +222,7 @@ def _parse_input(self, input):
                 value = ','.join(value)
                 input[key] = f'[{value}]'
-            elif 'mgmt_nw.iface' in key:
+            elif 'mgmt.interface' in key:
```
Yes

ypise commented on api/python/provisioner/params_spec.yaml:
```diff
@@ -26,29 +26,29 @@ network:
-  primary_data_netmask: cluster/srvnode-1/network/data_nw/netmask
-  primary_data_gateway: cluster/srvnode-1/network/data_nw/gateway
-  primary_data_network_iface: cluster/srvnode-1/network/data_nw/iface
+  primary_mgmt_ip: cluster/srvnode-1/network/mgmt/ipaddr
```
Yes

ypise commented on api/python/provisioner/params_spec.yaml:
```diff
@@ -59,7 +59,7 @@ release:
   _path: storage_enclosure.sls
-  controller_a_ip: storage_enclosure/controller/primary_mc/ip
+  controller_a_ip: storage_enclosure/controller/primary/ip
   controller_b_ip: storage_enclosure/controller/secondary_mc/ip
```
Yes

ypise commented on cli/src/auto-deploy:
```diff
@@ -384,19 +384,19 @@ update_pillar()
         echo "Updating interface names" 2>&1|tee -a $LOG_FILE
-        provisioner pillar_set cluster/srvnode-1/network/data_nw/iface [\"${pub_din}\",\ \"${pvt_din}\"]
-        provisioner pillar_set cluster/srvnode-2/network/data_nw/iface [\"${pub_din}\",\ \"${pvt_din}\"]
+        provisioner pillar_set cluster/srvnode-1/network/data/interface [\"${pub_din}\",\ \"${pvt_din}\"]
```
Yes

ypise replied:
Yes

ypise commented on cli/src/auto-deploy:
```diff
@@ -427,7 +427,7 @@ update_pillar()
     if [[ "$ctrl_a_ip_opt" == true ]]; then
         echo "Updating controller_ip1" 2>&1|tee -a $LOG_FILE
-        provisioner pillar_set storage_enclosure/controller/primary_mc/ip \"${ctrl_a_ip}\"
+        provisioner pillar_set storage_enclosure/controller/primary/ip \"${ctrl_a_ip}\"
```
Already is on line 436.
This script file is no more used.

ypise commented on cli/src/factory_ops/boxing/shutdown:
```diff
@@ -110,7 +110,7 @@ function shutdown_controller {
-    local ctrl_ip=$(get_pillar_data storage_enclosure:controller:primary_mc:ip)
+    local ctrl_ip=$(get_pillar_data storage_enclosure:controller:primary:ip)
     if [[ "$(salt-call grains.get id --output=newline_values_only)" == "srvnode-2" ]]; then
         ctrl_ip=$(get_pillar_data storage_enclosure:controller:secondary_mc:ip)
```
done

ypise commented on cli/src/factory_ops/unboxing/pre_unbox:
```diff
@@ -305,7 +305,7 @@ ctrl_ip_addr_get_set()
 
     echo "Updating storage enclosure pillar with new controller A IP address" | tee -a ${LOG_FILE}
-    _current_ip=$(grep -A1 primary_mc ${_storage_encl_sls_path} | tail -1 | awk '{ print $2 }')
+    _current_ip=$(grep -A1 primary ${_storage_encl_sls_path} | tail -1 | awk '{ print $2 }')
```
Done

ypise commented on pillar/components/cluster.sls:
```diff
@@ -16,67 +16,82 @@
-    hostname: srvnode-1
-    is_primary: true
+  server_nodes:
+    - srvnode_1
```
Yes, corrected

ypise commented on pillar/components/cluster.sls:
```diff
@@ -16,67 +16,82 @@
+  cluster_id:
+  storage_sets:
+    storage_set_1:
+      - srvnode_1
```
Corrected.

ypise commented on pillar/components/cluster.sls:
```diff
@@ -16,67 +16,82 @@
+    site_id:
+    storage_set_id:
+    node_id:
+    hostname: srvnode_1
```
Added the key

ypise commented on pillar/samples/config.ini:
```diff
@@ -3,39 +3,39 @@ search_domains=
-network.data_nw.netmask=
-network.data_nw.gateway=
-storage.metadata_device="/dev/sdb"
+network.mgmt.interface=eno1
```
corrected

ypise commented on pillar/samples/config.ini:
```diff
@@ -3,39 +3,39 @@ search_domains=
+network.mgmt.interface=eno1
+network.mgmt.netmask=
+network.mgmt.gateway=
+network.data.interface=enp175s0f0, enp175s1f0
```
corrected

ypise commented on pillar/samples/config.ini:
```diff
@@ -3,39 +3,39 @@ search_domains=
+network.mgmt.netmask=
+network.mgmt.gateway=
+network.data.interface=enp175s0f0, enp175s1f0
+network.data.netmask=
```
May be for netmask, but not for gateway IP

ypise commented on pillar/samples/config.ini:
```diff
@@ -3,39 +3,39 @@ search_domains=
+network.data.public_ip=172.20.30.103
 
 [storage-enclosure-default]
 type=RBOD
```
There should be no controller.
Corrected it.

ypise commented on pillar/samples/config.ini:
```diff
@@ -3,39 +3,39 @@ search_domains=
 [storage-enclosure-default]
 type=RBOD
-controller.primary_mc.ip=10.0.0.2
+controller.primary0.2
```
Removed controller.

ypise commented on pillar/samples/config.ini:
```diff
@@ -3,39 +3,39 @@ search_domains=
 type=RBOD
-controller.primary_mc.ip=10.0.0.2
+controller.primary0.2
 controller.secondary_mc.ip=10.0.0.3
```
removed controller

ypise commented on pillar/samples/config_vm.ini:
```diff
@@ -3,16 +3,16 @@ type=other
-network.data_nw.pvt_ip_addr=None
-network.data_nw.iface=eth1, eth2
-network.mgmt_nw.iface=eth0
+network.data.private_ip=None
```
Specifying public_ip would mean static IP config.  
No IP specified means DHCP config.

ypise commented on srv/_modules/sspl.py:
```diff
@@ -143,10 +143,10 @@ def _inject_storage_enclosure(config_dict={}):
+    config_dict['STORAGE_ENCLOSURE']['primary_controller_ip'] = storage_enclosure['primary']['ip']
+  if storage_enclosure['primary']['port']:
+    config_dict['STORAGE_ENCLOSURE']['primary_controller_port'] = storage_enclosure['primary']['port']
   if storage_enclosure['secondary_mc']['ip']:
```
Removed this code as it's irrelevant now (was relevant year and half back)

ypise commented on srv/components/sspl/config/commons.sls:
```diff
@@ -47,8 +47,8 @@ Add common config - BMC to Consul:
-        consul kv put storage_enclosure/controller/primary_mc/port {{ pillar['storage_enclosure']['controller']['primary_mc']['port'] }}
+        consul kv put storage_enclosure/controller/primary/ip {{ pillar['storage_enclosure']['controller']['primary']['ip'] }}
+        consul kv put storage_enclosure/controller/primary/port {{ pillar['storage_enclosure']['controller']['primary']['port'] }}
         consul kv put storage_enclosure/controller/secondary_mc/ip {{ pillar['storage_enclosure']['controller']['secondary_mc']['ip'] }}
```
Done

ypise commented:
@nitin-seagate 
Addressed review comments and made a few more changes.
The `is_primary` to '{"roles": ["primary"]}` change I am planning to deliver in a separate PR.

nitin-seagate commented on pillar/components/storage.sls:
```diff
@@ -15,15 +15,33 @@
+storage:
+    enclosure_1:
+      type: RBOD                      # RBOD/JBOD/Virtual/Other
+      primary:
```
Correct the key-path is
**storage>{enclosure-node}>controller>primary> . . .**

nitin-seagate commented on pillar/components/storage.sls:
```diff
@@ -15,15 +15,33 @@
         ip: 10.0.0.2
         port: 80
-      secondary_mc:
+      secondary:
```
storage>{enclosure-node}>controller>secondary> . . .

nitin-seagate commented on pillar/components/storage.sls:
```diff
@@ -15,15 +15,33 @@
+      secondary:
         ip: 10.0.0.3
         port: 80
       user: manage
```
**storage>{enclosure-node}>controller>secondary>user**

nitin-seagate replied:
storage>{enclosure-node}>controller>user
storage>{enclosure-node}>controller>secret

nitin-seagate commented on srv/components/csm/files/ha-params.tmpl:
```diff
@@ -1,11 +1,11 @@
-right-node: srvnode-2
\ No newline at end of file
+left-node: srvnode_1
+right-node: srvnode_2
```
Please add new line at EOF

nitin-seagate commented on srv/components/sspl/config/commons.sls:
```diff
@@ -35,21 +35,26 @@ Add common config - rabbitmq cluster to Consul:
-        consul kv put storage_enclosure/controller/user {{ pillar['storage_enclosure']['controller']['user'] }}
-        consul kv put storage_enclosure/controller/password {{ pillar['storage_enclosure']['controller']['secret'] }}
+{% for enclosure_id in enclosures %}
+        consul kv put storage/{{ enclosure_id }}/primary/ip {{ pillar['storage'][{{ enclosure_id }}]['primary']['ip'] }}
```
Correct key paths:
**pillar['storage'][{{ enclosure_id }}]['controller']['primary']['ip']
pillar['storage'][{{ enclosure_id }}]['controller']['primary']['port']
pillar['storage'][{{ enclosure_id }}]['controller']['secondary']['ip']
pillar['storage'][{{ enclosure_id }}]['controller']['secondary']['port']
pillar['storage'][{{ enclosure_id }}]['controller']['user']
pillar['storage'][{{ enclosure_id }}]['controller']['secret']**

nitin-seagate commented on srv/components/sspl/teardown/commons.sls:
```diff
@@ -44,27 +44,30 @@ Delete common config - rabbitmq cluster to Consul:
+    {% for node_id in server_nodes %}
+        consul kv delete bmc/{{ node_id }}/ip
+        consul kv delete bmc/{{ node_id }}/user
+        consul kv delete bmc/{{ node_id }}/secret['secret']
```
Is this correct?
secret['secret']

ypise replied:
Corrected

nitin-seagate commented on srv/components/sspl/teardown/commons.sls:
```diff
@@ -44,27 +44,30 @@ Delete common config - rabbitmq cluster to Consul:
-        consul kv delete storage_enclosure/controller/user
-        consul kv delete storage_enclosure/controller/password
+{% for enclosure_id in enclosures %}
+        consul kv delete storage/{{ enclosure_id }}/primary/ip
```
Key paths should be:
**consul kv delete storage/{{ enclosure_id }}/controller/primary ...
consul kv delete storage/{{ enclosure_id }}/controller/secondary ...**

nitin-seagate commented on srv/components/system/firewall/config.sls:
```diff
@@ -164,9 +164,9 @@ Management zone:
     - port_fwd:
       {% if pillar['cluster'][grains['id']]['is_primary'] %}
-      - {{ pillar['storage_enclosure']['controller']['primary_mc']['port'] }}:80:tcp:{{ pillar['storage_enclosure']['controller']['primary_mc']['ip'] }}
+      - {{ pillar['storage']['enclosure_1']['primary']['port'] }}:80:tcp:{{ pillar['storage']['enclosure_1']['primary']['ip'] }}
```
**controller** keyword missing?

nitin-seagate commented on srv/components/system/firewall/install.sls:
```diff
@@ -28,11 +28,6 @@ Unmask Firewall service:
       - Install firewalld
 
 # Ensure ssh works when the firwall servcie starts for the next time
-{% if 'mgmt0' in grains['ip4_interfaces'] and grains['ip4_interfaces']['mgmt0'] %}
```
Are we sure to delete this one?

ypise replied:
Yes, it is the network teaming format that we would require later for 2 network setup.

ypise replied:
Yes, as we are not using it anywhere.
The configuration here is addressing global zone and not targeted ones.

nitin-seagate commented on srv/components/system/network/mgmt/bonded/init.sls:
```diff
@@ -15,7 +15,7 @@
 #
 
-{%- if (pillar['cluster'][node]['network']['mgmt_nw']['ipaddr']) and (not pillar['cluster'][grains['id']]['network']['mgmt_nw']['gateway']) %}
+{%- if (pillar['cluster'][node]['network']['mgmt'ipaddr']) and (not pillar['cluster'][grains['id']]['network']['mgmt']['gateway']) %}
```
This needs to be corrected:
**['mgmt']['ipaddr']**

ypise replied:
fixed.
Should be: pillar['cluster'][node]['network']['mgmt']['public_ip']

nitin-seagate commented on test/api/python/provisioner/test_commands.py:
```diff
@@ -65,9 +65,12 @@ def storage_enclosure_data(mocker):
-                "primary_mc": {
+        "storage": {
+            "enclosure_1": {
+                "primary": {
```
**controller** keyword is missing?

ypise commented on Cortx-ProvisionerQuickstartGuide.md:
```diff
@@ -636,35 +630,35 @@ In such scenarios the destroy may get stuck somewhere due to some unknown reason
+              		mgmt:                  # Management network interfaces
+                		interfaces:	- eth0
+                		public_ip: 
                 		netmask: 255.255.255.0
```
Done

ypise commented on Cortx-ProvisionerQuickstartGuide.md:
```diff
@@ -636,35 +630,35 @@ In such scenarios the destroy may get stuck somewhere due to some unknown reason
+          		node_list: - srvnode_1
+          		srvnode_1:
+            		hostname: srvnode_1
             		is_primary: true
```
Will add it as a separate commit later.

ypise commented on Cortx-ProvisionerQuickstartGuide.md:
```diff
@@ -636,35 +630,35 @@ In such scenarios the destroy may get stuck somewhere due to some unknown reason
-                		ipaddr: 
+              		data:                  # Data network interfaces
+                		interfaces: - eth1
+                		public_ip: 
```
Done

ypise commented on cli/src/factory_ops/unboxing/pre_unbox:
```diff
@@ -296,16 +296,16 @@ ctrl_ip_addr_get_set()
-    #Update the storage_enclosure pillar with new controller IP.
-    _storage_encl_sls_path="${pillar_root}/uu_storage_enclosure.sls"
+    #Update the storage pillar with new controller IP.
+    _storage_encl_sls_path="${pillar_root}/uu_storage.sls"
```
user_updated_storage, used internally to manage pillar data.

ypise commented on cli/src/factory_ops/unboxing/system_check:
```diff
@@ -88,27 +88,27 @@ function check_hostname {
     # Only management IP would be tested as public data IP would be later updated during onboarding.
 
-    local mgmt_if=$(grep -m1 -A3 -P "mgmt_nw:" ${cluster_sls_path}|grep -A1 "iface:"|tail -1|cut -d'-' -f2|tr -d "[:space:]")
+    local mgmt_if=$(grep -m1 -A3 -P "mgmt:" ${cluster_sls_path}|grep -A1 "interface:"|tail -1|cut -d'-' -f2|tr -d "[:space:]")
```
Corrected.

ypise commented on pillar/components/cluster.sls:
```diff
@@ -16,69 +16,119 @@
+          roaming_ip: 192.168.0.23       # Applies to private data network
+      storage:
+        enclosure_id: enclosure_3
+        metadata_devices:              # Device for /var/motr and possibly SWAP
```
If you scroll up a bit you'll see that they still are.

ypise commented on pillar/samples/config_vm.ini:
```diff
@@ -3,16 +3,16 @@ type=other
-network.data_nw.pvt_ip_addr=None
-network.data_nw.iface=eth1, eth2
-network.mgmt_nw.iface=eth0
+network.data.private_ip=None
```
Not unless it's supposed to be configured as static IP.
We don't specify all the keys everywhere to avoid user jumbling.

ypise commented:
@nitin-seagate 
The final JSON export shall look like:
```
{
  "commons": {
    "health-map": {
      "path": "/opt/seagate/cortx_configs/healthmap/",
      "file": "healthmap-schema.json"
    },
    "version": {
      "consul": "1.7.8-1",
      "elasticsearch-oss": "6.8.8-1",
      "erlang": "latest",
      "kibana-oss": "6.8.8-1",
      "nodejs": "v12.13.0",
      "rabbitmq": "latest",
      "rsyslog": "8.40.0-1.el7",
      "rsyslog-elasticsearch": "8.40.0-1.el7",
      "rsyslog-mmjsonparse": "8.40.0-1.el7",
      "kafka": "2.13-2.7.0"
    }
  },
  "corosync-pacemaker": {
    "cluster_name": "cortx_cluster",
    "user": "hacluster",
    "secret": "gAAAAABgDbTrD1NlNbxpdxeJW4IrHr6edes-kTb19SH2nRPDMg0U6JJMqGlWVjdovDGAzYm1VCSxbRGNR0M6IZ0n97HEXOuvrRLCJO7I4PnGwds7uKuenK-kaK8oHL2Wqfl76a3rFGx7x1v__CLRweRLTARYi4GxynuSuwPFpZHLqwoAo8Xj92Hiosss2mck8oWbPekGfn5YY1VnufDHOxjOSa55bB6BTxGXWGfM5rQ_zgerVswJ_laua5inz6B4cIux2ggEczSIjp8QSziUwMX_ENijl65jI0kVhCALA1P4uTHccmwbJYvBYtJPQuIVrV8GaoHA0bv8A3XulIsagSk-MQGkrAS9qCInV9wFSgTBSxyWepFcGKR3vzGZrgZUcqOZzjgueR1KmFbKwjNKGvkKq0Hffi2AjhNQuq1PusE_bCHLQro15lHoCG_2aqemvJAtMk-KNDo75g9-vyZnOloXRMlDXr7WiEauKCo_-UdXxwf6nHbQbpSA66ApZd5ASc78BiE9HWHo4xHmPhMY5Oxe77SfqlpO0lsn_joNFhqu0F3Fc30DHOiV-N4Wmwv14iWewCY9i0mcud6aqfgFnJN4k_siGURe7Q=="
  },
  "elasticsearch": {
    "cluster_name": "elasticsearch_cluster",
    "network_host": "0.0.0.0",
    "http_port": 9200
  },
  "firewall": {
    "data_public": {
      "services": [
        "ssh",
        "high-availability"
      ],
      "ports": {
        "consul": [
          "8600/tcp",
          "8600/udp",
          "8500/tcp",
          "8301/tcp",
          "8301/udp",
          "8302/tcp",
          "8302/udp",
          "8300/tcp"
        ],
        "dhclient": [
          "68/udp"
        ],
        "dhserver": [
          "67/udp"
        ],
        "hare": [
          "8008/tcp"
        ],
        "nfs": [
          "2049/tcp",
          "2049/udp",
          "32803/tcp",
          "892/tcp",
          "875/tcp"
        ],
        "uds": [
          "5000/tcp",
          "5125/udp"
        ],
        "www": [
          "443/tcp"
        ],
        "s3": [
          "7081/tcp",
          "8081/tcp",
          "8082/tcp",
          "8083/tcp",
          "8084/tcp",
          "8085/tcp",
          "8086/tcp",
          "8087/tcp",
          "8088/tcp",
          "8089/tcp",
          "8090/tcp",
          "8091/tcp",
          "8092/tcp",
          "8093/tcp",
          "8094/tcp",
          "8095/tcp",
          "8096/tcp",
          "8097/tcp",
          "8098/tcp",
          "514/tcp",
          "514/udp",
          "8125/tcp",
          "6379/tcp",
          "9443/tcp",
          "9086/tcp"
        ]
      }
    },
    "mgmt_public": {
      "services": [
        "ssh",
        "high-availability",
        "ftp"
      ],
      "ports": {
        "consul": [
          "8600/tcp",
          "8600/udp",
          "8500/tcp",
          "8301/tcp",
          "8301/udp",
          "8302/tcp",
          "8302/udp",
          "8300/tcp"
        ],
        "csm": [
          "28100/tcp",
          "28101/tcp",
          "28102/tcp",
          "28103/tcp"
        ],
        "dhclient": [
          "68/udp"
        ],
        "elasticsearch": [
          "9200/tcp",
          "9300/tcp"
        ],
        "ntpd": [
          "123/udp"
        ],
        "openldap": [
          "389/tcp"
        ],
        "smtp": [
          "25/tcp"
        ],
        "saltmaster": [
          "4505/tcp",
          "4506/tcp"
        ],
        "uds": [
          "5000/tcp",
          "5125/udp"
        ],
        "www": [
          "443/tcp"
        ]
      }
    }
  },
  "haproxy": {
    "backend": {
      "s3authserver": {
        "ssl_enabled": false
      },
      "s3server": {
        "ssl_enabled": false
      }
    },
    "frontend": {
      "s3authserver": {
        "ssl_enabled": true
      },
      "s3server": {
        "ssl_enabled": true
      }
    },
    "nbproc": 12
  },
  "openldap": {
    "backend_db": "mdb",
    "root": {
      "user": "admin",
      "secret": "gAAAAABgDbTsfNurQu48F25y18tpzHnNn0Fedxhx9D0nijlEdRyRBNA1-OWaIBVtuZ07nOOFQRAtHGG_gPLDSilo0k7GElHAr8uzR6Rfo-3qKRmpp69422VOwsp0B02JceRcdjqrsQh6_pVuvi9gi5bTPwZyKViN-muuM29hy-py_Z2op-KqhcOGFK1SMNQJTRIQhC5AM6k6R3cAa07t_WiHKEqdygeHp97dx_qjvzcGdMppYEF7wGr79IOGexDFbdIIK685frCApIL5p-DnEmgah5WX36iu4MWEZIRN9X6hsvt_1dWkImLOWtO4fzvlWkebv65o6GgKy0MOcbdNXkJlf1mTQqXe35wxmT6XZ4xnnri8aSKqn_5B6md7dMAox3VDi3BAo_S-oOpbM_l4BVbKC0plDiHBO9hIg-fQnyJyw3w5L_MBkxTP8WDtp-Ch_bO7MKEahMRqbHzPXm7hWQtK2dXIzyQKvaqK7AGWdbXcPNO947gfBfdEp_BvHREYk_1zhK4s3QHHLVZ7qoJE433kOxN-JqeW6lYHHbr2cMHFBZhf31E9Ma_rOiZkDRnNf6wmMmNk-04AtFj9DXso4YZlU6ccgC4EHA=="
    },
    "sgiam": {
      "user": "admin",
      "secret": "gAAAAABgDbTs9Uz7Kz876524FV3iSjjaXBKz_8M7WfTCcRzZINeQCuMrhkyUH96eUEdjP3Ky0W-rmroNvfvKN7GhCXSPV3ISfV5d5yRn2XhNy-hV-P5WitzdS9pZuvSp3X0IU3bNVMgCZnI3EjgxgnhKqLozRwo1UnEEnJ57Lno_plVNJTGxm2sX7vbhw7fUs9fKzgYq9rDxsQYzhFo70s_HQNfk2VGNYtJmGekNv6Cqyz65A6wfweLnBynzlwkK9w3_FuWU87TI0jG4FRuGxRqIATaSMCI9rW8-X9ygvqnKVTIpjBgjD4HXLGVVwRJZD-xvlp8zQGVK7cEGwxuDUAGKYfque6lgWrl2GuZSgjykdkN9nYb44t3xdvhxbp8Vs9si5gc3ooS_uZV6hA5HAr_mg0WYT7UbCbE2q6sThB3CBmiZ6bYYdDf8mWch-zq33GZpBrYhO8wXMuTTdlVGhomSghEux6_8aCKuQb2CjQSZv5uNbjTRmw8ArwOQwcMdO6WKW1deVg-a19Yew9QrniY7SpRgTOGNFP1TBpHzzxtRJgYr6kStbID5YG8NCUuEbeM4S8TAEhe5hnF4nfrTPPYhUuukNGH1gg=="
    }
  },
  "rabbitmq": {
    "cluster_nodes": "localhost",
    "erlang_cookie": "QLDZYPYEYGHECTHYQXFJ"
  },
  "release": {
    "product": "LR2",
    "setup": "cortx",
    "type": "bundle",
    "target_build": "http://cortx-storage.colo.seagate.com/releases/cortx/github/integration-custom-ci/release/centos-7.8.2003/custom-build-486/",
    "update": {
      "base_dir": "/opt/seagate/cortx/updates",
      "repos": {}
    }
  },
  "rsyslog": {
    "host": "localhost",
    "port": 514
  },
  "storage": {
    "enclosure-1": {
      "type": "RBOD",
      "controller": {
        "type": null,
        "primary": {
          "ip": "10.0.0.2",
          "port": 80
        },
        "secondary": {
          "ip": "10.0.0.3",
          "port": 80
        },
        "user": "manage",
        "secret": "gAAAAABgDbTt1kjn3MuL_r6w9DuqvKdKdeVjyykBQ-fJPUYDrE76ZUJ49Wl4Y8ocvXOq07nU7JHnjZ7Y6dFJ99rog1rf5FtkLl1bFfjLi_YJMO4ZUuq4nbRrPkVgT-_UgIx4qri2WPj-iVjdDacq56qBzrK11BqW1pzdj_FB7MkjY2cIaHAF-VAqh7OHeVbqIlqPFD8tUMvwJlvK_wJCnifzxYgNYQnfW-KY7oC_xrOu4vS5TjjUR1t5eS31SOXHlJYE5qqmjvcFC2p56Dew4hJIP7bhPV_TqKG8Pj2a7r1wReligRRNPoQRv1r4szcaTqng1IWeX1ARWE4OQMMvZBI2-1EMGRhePW_UAXvFU1nklwicWioFJVzpCfK9KuDsJCWgOq-6wDbMFAEEXAi1m7Fo9JAYNCXrFb_oMt2UuA_omt4r3BT4WCeBvk5Vy1JPt18vV2Pvb1lZ8P1tZDNVuedZMgokqWpEDgKSEP58U04JsugOnJ9GHesqQ1w7zvlkfgcVutum1eDpGI8gaSenN0OjuDRYfHcf-UZKtsqCMD6XuVsP-urJLeiM7PfeSCs3ZCnDrWoYZRl0wGPg6aoaG4HdFGyQxsLaWw=="
      }
    }
  },
  "system": {
    "ntp": {
      "time_server": "time.seagate.com",
      "time_zone": "UTC"
    },
    "service_user": {
      "name": "cortxub",
      "secret": "gAAAAABgDbTtAZepxgUIFTDh4Cf31AinZNt-7v_LuI4ChbeDgHoonrHgRkcya2A-uQwtIA1QmLT8Dl2Q8jwXMsRLhiwW5i4jXg==",
      "groups": [
        "wheel"
      ],
      "shell": "/usr/bin/bash"
    },
    "cortx_user": {
      "name": "cortxcli",
      "secret": "gAAAAABgDbTuhgNw8Y3k6u4t7T6y7b2-xbyj838oPTTXlg9KMJaRhFaLGOb7zk4KqdgfX_SDtWFne8Rp4bzZOuFTU1y7-z9qrCKJnYcw_2V6jg3S80Joa391c-EUrBwAmU1JXJspBrZNW7mLRnJJKUyjvlgqfupjzwPoZGnO80riKzFVHHUBqLO0jsg5bJuV2xQTL5QEQJfyEkdfgsuHCNyxRqTufptG-PUM4CLv00Rz01bx5Yl849ZIZ3qd-xQurq3Xvu9rI8koTIuMV8VNX-mNCYwznKQzk8EydcujiW-Okk9iUtf25GLMwRk_n7W98fcgLxVIFFjl-_7kJpTsA3GcwqRDkZjejzdCgBVoT4fwfuIxZlhI425ExarYfxvecmfHD5IG4RJ9xkYKR7_jampk6ZOgX3d4aUxlAxGgm38jtA-YXojhYqD22bhuAutnUv2HpAt7HPiagcVk-EPucrQMY6rA3DwFcG8Gcr70wzgBO4Mo7WxJAP0pCPbw62V6QKUgpSDSD82GZj-t5wMgg321c_FLu1ZsUT7uWHaurVVNhmKfTFSCSDi5zMMNuEagAJTDfJtYC7iWdKNNDdE66ZmSCwLgDiVtfQ=="
    }
  },
  "cluster": {
    "cluster_id": "868ac031-e3f7-4c2e-ae45-76f6d5c81c6a",
    "storage_sets": {
      "storage-set-1": [
        "srvnode-1"
      ]
    },
    "search_domains": null,
    "dns_servers": null,
    "server_nodes": {
      "7c4fd75dfedd7662e6a39b0a53274922": "srvnode-1"
    },
    "srvnode-1": {
      "rack_id": null,
      "site_id": null,
      "storage_set_id": "storage-set-1",
      "node_id": null,
      "machine_id": "7c4fd75dfedd7662e6a39b0a53274922",
      "hostname": "ssc-vm-2132.colo.seagate.com",
      "roles": [
        "primary",
        "openldap_server",
        "kafka_server"
      ],
      "is_primary": true,
      "bmc": {
        "ip": null,
        "user": "",
        "secret": "gAAAAABgDbUIdwF_RhvzvBa13hZLDylfYMpMDW7LlH_IacFFdRLNfPP9xdmUkxY6zyvvNhPBec048dP_3mD3y27u7eQZsFyI6Q=="
      },
      "network": {
        "mgmt": {
          "interfaces": [
            "eth0"
          ],
          "public_ip": null,
          "netmask": null,
          "gateway": null
        },
        "data": {
          "public_interfaces": [
            "eth1",
            "eth2"
          ],
          "private_interfaces": [
            "eth3",
            "eth4"
          ],
          "interface_type": "tcp",
          "transport_type": "lnet",
          "public_ip": null,
          "private_ip": "",
          "netmask": null,
          "gateway": null,
          "roaming_ip": "127.0.0.1"
        }
      },
      "storage": {
        "enclosure_id": "enclosure-1",
        "metadata_devices": [
          "/dev/sdb"
        ],
        "data_devices": [
          "/dev/sdc"
        ]
      },
      "s3_instances": 11
    },
    "*": {
      "network": {
        "data": {
          "roaming_ip": "127.0.0.1"
        }
      },
      "storage": {
        "data_devices": [
          "/dev/sdc"
        ],
        "metadata_devices": [
          "/dev/sdb"
        ]
      }
    }
  },
  "s3server": {
    "no_of_inst": 1
  }
}
```  
The null and integers shall be addressed in PR: #843
